doc/rtd/explanation/about-cloud-config.rst

```diff
@@ -0,0 +1,85 @@
+.. _about-cloud-config:
+
+About the cloud-config file
+***************************
+
+The ``#cloud-config`` file is a type of user data that cloud-init can consume
+to automatically set up various aspects of the system.
+
+It overrides all other types of user data, so if there is a conflict between
+the config you specify in other :ref:`user data formats <user_data_formats>`,
+the contents of the ``#cloud-config`` file will take precendence.
+
+Note that networks are not configurable via the ``#cloud-config`` file because
+:ref:`network configuration <network_config>` occurs before user data is
+consumed and applied.
+
+How do I create a cloud-config file?
+====================================
+
+The cloud-config file can be written in any valid YAML but the first line
+**must start** with ``#cloud-config``. This line identifies the file to
+cloud-init and ensures that it will be processed as intended.
+
+After the first line, every aspect of the system's configuration is controlled
+through specific cloud-init **modules**. Each module included in the
+``cloud-config`` file can be thought of as a section.
```

Comment:
I think one of the reasons this is hard to document is that we use a lot of terminology quite loosely/interchangeably as a shorthand without it being rigorously defined. What's the difference between a top level key and a top level config module?

Reply:
Nevermind, I think I answered my own question :upside_down_face:

Reply:
no prob. to clarify, a config module in cloud-init can take action based on the presence of `0 to n` top-level cloud-config keys. Each module defines the cloud-config keys that will activate this module as documented in the module reference page in the `Activate only on keys: ansible` metadata associated with that module. Some examples:
- The [Bootcmd module is only run if the top-level key `bootcmd` is present.](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#bootcmd)
- The  [Disk setup module is run if either ``disk_setup`` or ``fs_setup`` are present](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#disk-setup)
- The [Locale config module does not define a limited activate on schema keys, so it will be run whether or not a `locale` key is present in cloud-config or not](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#locale)